### PR TITLE
enh(configuration): remove nagios_path_img option

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

Since Centreon no longer supports Nagios, it is no longer useful to copy images from Centreon to a third-party folder.

This mechanism was half implemented that could occurs errors on pollers/remote conf export. (e.g moveFiles script try to recreate already existing images)

We choose to remove this option from Centreon and replace it by the media path of centreon.

**Fixes** # MON-19073

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [x] 23.04.x
- [x] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Non-Regression on image upload, configuration export, create service / host with icons

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by dmyios — APPROVED** [review]

**Review by dmyios — DISMISSED** [review]

**Review by abuathier — APPROVED** [review]

File: centreon/www/install/php/Update-23.04.1.php [hunk 1]
```diff
@@ -18,3 +18,40 @@
...
+$removeNagiosPathImg = function(CentreonDB $pearDB) {
+    $selectStatement = $pearDB->query("SELECT 1 FROM option WHERE `key`='nagios_path_img'");
+    if($selectStatement->rowCount() > 0) {
+        $pearDB->query("DELETE FROM option WHERE `key`='nagios_path_img'");
+    }
+};
+
+try {
+    // Transactional queries
+    $pearDB->beginTransaction();
```
Comment (callapa) [inline centreon/www/install/php/Update-23.04.1.php:38]:
```suggestion
    if (! $pearDB->inTransaction()) {
        $pearDB->beginTransaction();
    }
```

File: centreon/www/install/php/Update-23.10.0.php [hunk 2]
```diff
@@ -18,3 +18,40 @@
...
+$removeNagiosPathImg = function(CentreonDB $pearDB) {
+    $selectStatement = $pearDB->query("SELECT 1 FROM option WHERE `key`='nagios_path_img'");
+    if($selectStatement->rowCount() > 0) {
+        $pearDB->query("DELETE FROM option WHERE `key`='nagios_path_img'");
+    }
+};
+
+try {
+    // Transactional queries
+    $pearDB->beginTransaction();
```
Comment (callapa) [inline centreon/www/install/php/Update-23.10.0.php:38]: If you start a new SQL transaction when another one has already been started, the previous transaction will be automatically committed.
```suggestion
    if (! $pearDB->inTransaction()) {
        $pearDB->beginTransaction();
    }
```

File: centreon/www/class/CentreonLegacyConstants.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+class CentreonLegacyConstants
+{
+    public const CENTREON_MEDIA_PATH = _CENTREON_PATH_ . '/www/img/media/';
```
Comment (callapa) [inline centreon/www/class/CentreonLegacyConstants.php:24]: Why not put this constant in the **centreon/www/class/centreonMedia.class.php** classs?

Comment (jeremyjaouen) [reply centreon/www/class/CentreonLegacyConstants.php]: indeed !

**Review by dmyios — APPROVED** [review]

**Review by abuathier — APPROVED** [review]

**Pull request merged**